python/engine/config.py

```
@@ -161,6 +172,19 @@ def _validate(self):
         if self.memory_reset not in ("carry_forward", "periodic"):
             raise ValueError(
                 f"memory_reset must be 'carry_forward' or 'periodic', got '{self.memory_reset}'")
+        if self.gate_warmup_theta_floor_init is not None:
+            if len(self.gate_warmup_theta_floor_init) != self.k:
+                raise ValueError(
+                    f"gate_warmup_theta_floor_init length "
+                    f"{len(self.gate_warmup_theta_floor_init)} must match k={self.k}")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)